File: tool4_junit/codeFiles/python/Q5_list.py
import unittest

class Q5List:
    def __init__(self):
        self.items = []

    def find(self, key):
        return key in self.items

    def insert(self, data):
        self.items.append(data)


class TestQ5List(unittest.TestCase):
    def test_find_insert(self):
        ls = Q5List()

        self.assertFalse(ls.find(42))
        self.assertFalse(ls.find(10))

        ls.insert(42)
        self.assertTrue(ls.find(42))
        self.assertFalse(ls.find(10))

        ls.insert(10)
        self.assertTrue(ls.find(42))
        self.assertTrue(ls.find(10))

if __name__ == '__main__':
    unittest.main()

File: tool4_junit/codeFiles/junit/src/junit/Q5_list.java
package junit;

import java.util.ArrayList;

public class Q5_list {
	 private ArrayList<Integer> items;

	    public Q5_list() {
	        this.items = new ArrayList<Integer>();
	    }

	    public boolean find(int key) {
	        return items.contains(key);
	    }

	    public void insert(int data) {
	        items.add(data);
	    }
}
